lib/collection/src/shards/local_shard/scroll.rs

```diff
@@ -303,7 +304,7 @@ impl LocalShard {
 
         let ordered_records = point_ids
             .iter()
-            .filter_map(|point| records_map.get(point).cloned())
+            .filter_map(|point_id| records_map.get(point_id).cloned())
```

Comment:
why not do the same trick here?

Reply:
We do return the same point ID multiple times there. It happens if you have a point with multiple payload values, and it returns it for each value.

On the first push I actually used the same trick here, and tests started failing because of the above.

Reply:
Makes sense :+1: